server/controllers/chartsController/index.js

```diff
@@ -1,80 +1,156 @@
+        const totalsDataKey = `${TOTALS_STUDY}-${label}`
+        const isVariableValueEmpty = value === EMPTY_VALUE
+        const shouldCountSubject = isVariableValueEmpty
+          ? subjectDayData.every((day) => day[chart.variable] === value)
+          : subjectDayData.some((dayData) => dayData[chart.variable] == value)
 
-      labelMap.set(label, { name: label, color })
-      processData({ shouldCountSubject, dataMap, dataKey, totalsDataKey })
-      processTotals({
-        shouldCountSubject,
-        studyTotals,
-        siteName,
-        targetValue,
+        labelMap.set(label, { name: label, color })
+        processData({ shouldCountSubject, dataMap, dataKey, totalsDataKey })
+        processTotals({
+          shouldCountSubject,
+          studyTotals,
+          siteName,
+          targetValue,
+        })
       })
-    })
-  }
+    }
+
+    const dataBySite = postProcessData(dataMap, studyTotals)
 
-  const dataBySite = postProcessData(dataMap, studyTotals)
+    if (isAnyTargetIncluded(studyTotals)) {
+      labelMap.set(N_A, { name: N_A, color: '#808080' })
+    }
 
-  if (isAnyTargetIncluded(studyTotals)) {
-    labelMap.set(N_A, { name: N_A, color: '#808080' })
+    return {
+      chart,
+      dataBySite: Array.from(dataBySite.values()),
+      labels: Array.from(labelMap.values()),
+      studyTotals,
+      filters,
+    }
+  } else {
+    return {
+      chart,
+      dataBySite: [],
+      labels: Array.from(labelMap.values()),
+      studyTotals,
+      filters,
+    }
   }
+}
 
-  return {
-    chart,
-    dataBySite: Array.from(dataBySite.values()),
-    labels: Array.from(labelMap.values()),
-    studyTotals,
+const getAllSubjects = async ({
+  dataDb,
+  chart,
+  userAccess,
+  areFiltersRequested,
+  filters,
+}) => {
+  if (!areFiltersRequested) {
+    return await dataDb
+      .collection(collections.toc)
+      .find(
+        {
+          assessment: chart.assessment,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+        { projection: { collection: 1, study: 1, _id: 0 } }
+      )
+      .toArray()
+  } else {
+    const mongoFilterQuery = mongoQueryFromFilters(filters)
+    const filteredSubjects = []
+    const criteriaCursor = await dataDb.collection(collections.toc).find(
+      {
+        assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+        study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+      },
+      {
+        projection: { collection: 1, study: 1, _id: 0, subject: 1 },
+      }
+    )
+
+    for await (let doc of criteriaCursor) {
+      const data = await dataDb
+        .collection(doc.collection)
+        .findOne({ $or: mongoFilterQuery })
+
+      if (data) filteredSubjects.push(data?.subjectid)
+    }
+
+    return await dataDb
+      .collection(collections.toc)
+      .find(
+        {
+          assessment: chart.assessment,
+          subject: { $in: filteredSubjects },
+        },
+        { projection: { collection: 1, study: 1, _id: 0 } }
```

Comment:
can we DRY up some of these options? It seems like we could extract constants for default options being the assessment and projection.